homeassistant/components/tado/config_flow.py

```diff
@@ -110,6 +112,25 @@ async def async_step_homekit(
         self._abort_if_unique_id_configured()
         return await self.async_step_user()
 
+    async def async_step_import(self, import_config: dict[str, Any]) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+        _LOGGER.debug("Importing Tado from configuration.yaml")
+        self._async_abort_entries_match(
+            {
+                CONF_USERNAME: import_config[CONF_USERNAME],
+                CONF_PASSWORD: import_config[CONF_PASSWORD],
+                CONF_HOME_ID: import_config.get(CONF_HOME_ID, None),
+            }
+        )
+        return self.async_create_entry(
```

Comment:
Validate the config as we do in the user step before creating the entry. Instead of showing an error in the form, abort the flow. If the flow is aborted due to a problem we should create a repair issue about that.

Reply:
Validation added.